src/jquery.i18n.messagestore.js

```diff
@@ -111,13 +112,17 @@
 	function jsonMessageLoader( url ) {
 		var deferred = $.Deferred();
 
-		$.getJSON( url )
-			.done( deferred.resolve )
-			.fail( function ( jqxhr, settings, exception ) {
-				$.i18n.log( 'Error in loading messages from ' + url + ' Exception: ' + exception );
-				// Ignore 404 exception, because we are handling fallabacks explicitly
-				deferred.resolve();
-			} );
+		$.ajax( {
+			url: url,
+			dataType: 'json',
+			async: !sync
```

Comment:
synchronous loading of any resource is very bad from performanace and user experience perspective. If used with a complex application, the implications are very random and unexpected. 
So, if there is a usecase that is not really supported by this library, we can discuss alternate approaches but this approach is problematic